src/test/java/org/greenbuttonalliance/gbaresourceserver/customer/repository/CustomerAccountRepositoryTest.java

```diff
@@ -0,0 +1,297 @@
+
+		Assertions.assertEquals(
+			findByAllSize,
+			testDataSize,
+			() -> String.format("findByAll size of %s does not match test data size of %s", findByAllSize, testDataSize)
+		);
+	}
+
+	@Test
+	public void entityMappings_areNotNull() {
+		CustomerAccount fullyMappedCustomerAccount = customerAccountRepository.findById(UuidCreator.getNameBasedSha1(UuidCreator.NAMESPACE_URL, PRESENT_TITLE)).orElse(null);
+		Assumptions.assumeTrue(fullyMappedCustomerAccount != null);
+
+		// first-level mappings; directly from fullyMappedCustomerAccount
+		Function<CustomerAccount, Optional<ElectronicAddress>> customerAccountToElectronicAddress = ca -> Optional.ofNullable(ca.getElectronicAddress());
+		Function<CustomerAccount, Optional<Status>> customerAccountToDocStatus = ca -> Optional.ofNullable(ca.getDocStatus());
+		Function<CustomerAccount, Optional<Status>> customerAccountToStatus = ca -> Optional.ofNullable(ca.getStatus());
+		Function<CustomerAccount, Optional<Organisation>> customerAccountToContactInfo = ca -> Optional.ofNullable(ca.getContactInfo());
+		Function<CustomerAccount, Optional<Set<AccountNotification>>> customerAccountToAccountNotifications = ca -> Optional.ofNullable(ca.getAccountNotifications());
+
+		// second-level mappings; directly from fullyMappedCustomerAccount.contactInfo
+		Function<CustomerAccount, Optional<StreetAddress>> customerAccountToOrganisationStreetAddress = customerAccountToContactInfo.andThen(opt -> opt.map(
+			Organisation::getStreetAddress
+		));
+		Function<CustomerAccount, Optional<StreetAddress>> customerAccountToOrganisationPostalAddress = customerAccountToContactInfo.andThen(opt -> opt.map(
+			Organisation::getPostalAddress
+		));
+		Function<CustomerAccount, Optional<TelephoneNumber>> customerAccountToOrganisationPhone1 = customerAccountToContactInfo.andThen(opt -> opt.map(
+			Organisation::getPhone1
+		));
+		Function<CustomerAccount, Optional<TelephoneNumber>> customerAccountToOrganisationPhone2 = customerAccountToContactInfo.andThen(opt -> opt.map(
+			Organisation::getPhone2
+		));
+		Function<CustomerAccount, Optional<ElectronicAddress>> customerAccountToOrganisationElectronicAddress = customerAccountToContactInfo.andThen(opt -> opt.map(
+			Organisation::getElectronicAddress
+		));
+
+		// third-level mappings; directly from fullyMappedCustomerAccount.contactInfo.streetAddress
+		Function<CustomerAccount, Optional<StreetDetail>> customerAccountToOrganisationStreetAddressStreetDetail = customerAccountToOrganisationStreetAddress.andThen(opt -> opt.map(
+			StreetAddress::getStreetDetail
+		));
+		Function<CustomerAccount, Optional<TownDetail>> customerAccountToOrganisationStreetAddressTownDetail = customerAccountToOrganisationStreetAddress.andThen(opt -> opt.map(
+			StreetAddress::getTownDetail
+		));
+		Function<CustomerAccount, Optional<Status>> customerAccountToOrganisationStreetAddressStatus = customerAccountToOrganisationStreetAddress.andThen(opt -> opt.map(
+			StreetAddress::getStatus
+		));
```

Comment:
It shouldn't really be the responsibility of this class to test the object mappings past the first level, but similar to how the mapping tests in `IntervalBlockRepositoryTest` went all the way down to `ReadingQuality`, we're not going to have an `OrganisationRepositoryTest` to test the mappings of `Organisation`, and I wanted to make sure those mappings were tested somewhere instead of completely left out from the codebase.

Reply:
Sounds reasonable.